server/src/main/java/org/opensearch/cluster/service/MasterThrottlingStats.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.service;
+
+import org.opensearch.common.metrics.CounterMetric;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Contains stats of Master Task Throttling.
+ * It stores the total cumulative count of throttled tasks per task type.
+ */
+public class MasterThrottlingStats {
+
+    private Map<String, CounterMetric> throttledTasksCount = new ConcurrentHashMap<>();
+
+    public void incrementThrottlingCount(String type, final int permits) {
+        if (!throttledTasksCount.containsKey(type)) {
+            throttledTasksCount.put(type, new CounterMetric());
```

Comment:
This will not be thread safe, if multiple threads call the containsKey, could override each other with put call. You can use computeIfAbsent instead.

Reply:
Sure, corrected it.